### PR TITLE
Gem Blog App - processing data in models

### PR DESCRIPTION
**Pull request by tomdan-ai**
### Create models
-  Create model classes for all entities as shown in the [ERD diagram](../images/blog_app_erd_v1_1.png).
-  Set up associations between models.
    - Remember that `author_id` column in `posts` table should be the foreign_key for the `users` table.

### Use models to insert data
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user
- Create at least 4 posts written by one of the users you created 
- Create at least 6 posts comments for one of the posts you created by running the following code:
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
-# rubocop:disable all
-
 class Like < ApplicationRecord
-  belongs_to :user
   belongs_to :post
+  belongs_to :user
```
Comment (julie-ify) [inline app/models/like.rb]: - [ ] The like should belong to an `:author`, not `:user`. Kindly fix that.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+  after_initialize :set_defaults
+
+  private
+
+  def set_defaults
+    self.postsCounter ||= 0
+  end
 end
```
Comment (julie-ify) [inline app/models/user.rb]: - [ ] The User model should have a method for returning the 3 most recent posts.

  ```
  def recent_posts
    posts.order('created_at Desc').limit(3)
  end
  ```

File: app/models/post.rb [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
+
+  def update_posts_counter
+    author.increment!(:postsCounter)
+  end
+
+  def set_defaults
+    self.likes_counter ||= 0
+    self.comments_counter ||= 0
+  end
 end
```
Comment (julie-ify) [inline app/models/post.rb]: - [ ] The Post model should have a method for returning the 5 most recent comments.

  ```
  def recent_comments
    comments.order(created_at: :desc).limit(5)
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @tomdan-ai,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @tomdan-ai, 👋 👋  

Your project is complete! Great job for following the project requirement🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Blog App - processing data in models implemented ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**